packages/ui/src/components/Visualization/Custom/hooks/paste-step.hook.tsx

```diff
@@ -0,0 +1,40 @@
+import { useCallback, useContext, useMemo } from 'react';
+import { AddStepMode, IVisualizationNode } from '../../../../models/visualization/base-visual-entity';
+import { EntitiesContext } from '../../../../providers/entities.provider';
+import { ClipboardManager } from '../../../../utils/ClipboardManager';
+import { DefinedComponent } from '../../../../models/camel-catalog-index';
+
+export const usePasteStep = (
+  vizNode: IVisualizationNode,
+  mode:
+    | AddStepMode.InsertChildStep
+    | AddStepMode.InsertSpecialChildStep
+    | AddStepMode.AppendStep = AddStepMode.AppendStep,
+) => {
+  const entitiesContext = useContext(EntitiesContext);
+
+  const onPasteStep = useCallback(async () => {
+    if (!vizNode || !entitiesContext) return;
+
+    try {
+      const pastedNodeValue = ClipboardManager.paste();
+      if (pastedNodeValue) {
+        /** Add new node to the entities */
+        vizNode.addBaseEntityStep(pastedNodeValue as DefinedComponent, mode);
```

Comment:
Basically, here we are casting clipboard content of type `IClipboardCopyObject` to `DefinedComponent`!

Reply:
Ywa, but unfortunately this is not type-safe, so this could :boom:

Reply:
I agree. and this way, it doesn't also works with other entities type as well so I need to change this anyways.